includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -531,12 +555,45 @@ private function get_courses( $args ) {
 		if ( isset( $args['search'] ) ) {
 			$course_args['s'] = $args['search'];
 		}
-
+		add_filter( 'posts_clauses', [ $this, 'add_lesson_completion_count_to_course_query' ] );
 		// Using WP_Query as get_posts() doesn't support 'found_posts'
-		$courses_query     = new WP_Query( apply_filters( 'sensei_analysis_overview_filter_courses', $course_args ) );
+		$courses_query = new WP_Query( apply_filters( 'sensei_analysis_overview_filter_courses', $course_args ) );
+		remove_filter( 'posts_clauses', [ $this, 'add_lesson_completion_count_to_course_query' ] );
 		$this->total_items = $courses_query->found_posts;
 		return $courses_query->posts;
+	}
+	/**
+	 * Add count of completed lessons per course for all the active students.
+	 *
+	 * @since  4.2.0
+	 * @access public
+	 *
+	 * @param array $clauses Associative array of the clauses for the query.
+	 *
+	 * @return array Modified associative array of the clauses for the query.
+	 */
+	public function add_lesson_completion_count_to_course_query( $clauses ) {
+		global $wpdb;
+
+		$clauses['fields'] .= ", COUNT( {$wpdb->comments}.comment_approved ) as completed_lesson_count";
+		// Get postmeta rows that have meta_value of course ID and meta_key of '_lesson_course' to get relation
+		// between lessons and the course.
+		$clauses['join'] .= " LEFT JOIN {$wpdb->postmeta} as lessons ON lessons.meta_value = {$wpdb->posts}.ID";
```

Comment:
It's a nice practice to alias the `postmeta` as `pm` or `pm1` if you have multiple joins. The current name is a bit confusing to me.

Reply:
jup, changed it. [Here](https://github.com/Automattic/sensei/pull/4873/commits/06d86139fde8614c5d888b8b4c45f4e850eb18f4#diff-6ce1dc075c4832d5bf050e621bcfecbab91d97504e8bc565f4f476b1ea2f654dR582) and [here](https://github.com/Automattic/sensei/pull/4873/commits/06d86139fde8614c5d888b8b4c45f4e850eb18f4#diff-6ce1dc075c4832d5bf050e621bcfecbab91d97504e8bc565f4f476b1ea2f654dR590).